sql/src/test/java/io/crate/expression/scalar/cast/CastFunctionTest.java

```diff
@@ -42,12 +43,19 @@ public void testNormalize() throws Exception {
         assertNormalize("cast(name as long)", isFunction("to_long"));
     }
 
+    @Test
+    public void testInvalidCast() {
+        expectedException.expect(ConversionException.class);
+        expectedException.expectMessage("Cannot cast 10.4 to type long");
+        assertEvaluate("cast(10.4 as long)", 10L);
```

Comment:
I think explicit down-casts should still be allowed. It would also be inconsistent as we probably allow cast(double_col as long)`

Reply:
Good point. Will push a change to allow downcasts with the cast function.